lfs/lfs.go

```diff
@@ -98,13 +100,19 @@ func LinkOrCopyFromReference(cfg *config.Configuration, oid string, size int64)
 	if cfg.LFSObjectExists(oid, size) {
 		return nil
 	}
-	altMediafile := cfg.Filesystem().ObjectReferencePath(oid)
+	altMediafiles := cfg.Filesystem().ObjectReferencePaths(oid)
 	mediafile, err := cfg.Filesystem().ObjectPath(oid)
 	if err != nil {
 		return err
 	}
-	if altMediafile != "" && tools.FileExistsOfSize(altMediafile, size) {
-		return LinkOrCopy(cfg, altMediafile, mediafile)
+	for _, altMediafile := range altMediafiles {
+		tracerx.Printf("altMediafile: %s", altMediafile)
+		if altMediafile != "" && tools.FileExistsOfSize(altMediafile, size) {
+			err = LinkOrCopy(cfg, altMediafile, mediafile)
+			if err == nil {
+				break
```

Comment:
Maybe we could `return err` here, and `return nil` at the end of the function? I think that makes it a bit more explicit what's happening here

Reply:
Thanks for the suggestion, I think that the existing implementation might be a little clearer. The loop is trying to call `LinkOrCopy()` with each alternate to find the contents of an object and returning the last error.

When we get a nil error here, there's nothing left to do (i.e., we found an alternate that has the right objects, so we can stop going further, hence `break`-ing the loop).

I think it's prudent to keep the `return err` at the end of the function, since it indicates that we'll return the last-returned error from the loop. IIRC, it would be incorrect to write `return nil` at the end of the function, since `err` is not guaranteed to be non-`nil` by the end of the function.

Reply:
Ah okay! That makes sense to me, in that case I agree we should leave this as-is